client/components/auto-direction/index.jsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import userModule from 'lib/user';
+
+const user = userModule();
+
+const getContent = ( { props: reactComponentProps } ) => {
+	// The child is a text node
+	if ( typeof reactComponentProps.children === 'string' ) {
+		return reactComponentProps.children;
+	}
+
+	// This child has it's content set to external HTML
+	if ( typeof reactComponentProps.dangerouslySetInnerHTML === 'object' ) {
+		// Strip tags because we're only interested in the text, not markup
+		// copied from: http://stackoverflow.com/questions/5002111/javascript-how-to-strip-html-tags-from-string#answer-5002161
+		return reactComponentProps.dangerouslySetInnerHTML.__html
+			? reactComponentProps.dangerouslySetInnerHTML.__html.replace( /<\/?[^>]+(>|$)/g, '' )
+			: '';
+	}
+
+	// This child is some kind of input
+	if ( typeof reactComponentProps.value === 'string' ) {
+		return reactComponentProps.value;
+	}
+
+	// We have no idea how to get this element's content or it's not a leaf component
+	return null;
+};
+
+// Copied from: https://github.com/twitter/RTLtextarea/blob/master/src/RTLText.module.js#L46
```

Comment:
you'll need to add https://github.com/twitter/RTLtextarea/blob/master/LICENSE to https://github.com/Automattic/wp-calypso/blob/master/CREDITS.md